configure.ac

```diff
@@ -258,6 +258,50 @@ fi
 #
 # libsecp256k1
 #
+AC_ARG_WITH([system-secp256k1],
+    [AS_HELP_STRING([[--with-system-secp256k1[=PKG]]],
+        [build using system-installed libsecp256k1 instead of bundled, passing PKG (default: libsecp256k1 or libsecp256k1_zkp, depending on --enable-standard-secp) to pkg-config (default: no)])],
+    [AS_IF([test "x$withval" = xyes],
+        [AM_COND_IF([BUILD_STANDARD_SECP], [with_system_secp256k1=libsecp256k1], [with_system_secp256k1=libsecp256k1_zkp])])],
+    [with_system_secp256k1=no])
+AM_CONDITIONAL([LINK_SYSTEM_SECP256K1], [test "x$with_system_secp256k1" != xno])
+AM_COND_IF([LINK_SYSTEM_SECP256K1], [
+    saved_LIBS=$LIBS
+    PKG_CHECK_MODULES([libsecp256k1], [$with_system_secp256k1])
+    LIBS="$libsecp256k1_LIBS $LIBS"
+    AM_COND_IF([BUILD_STANDARD_SECP], [
+        AC_CHECK_FUNCS([
```

Comment:
I think this might be cleaner if its broken into:

- check for the standard modules (for all configs)
- if non-standard secp, check for ecdsa-s2c
- if elements, check for generator, rangeproof, surjectionproof, and whitelist

This will be easier to update going forward when new modules are required.

Reply:
The only reason I didn't do it that way was to improve the error messages. The first error encountered will abort the configure, yet the message will not report all needed modules. (This is true presently, but your suggestion makes it even worse.) But I agree in principle. I will pursue a cleaner way of checking for all needed modules at once and reporting all in a single error message by first building lists of needed modules, depending on enabled options, before performing the check.

Reply:
Thanks. As a maintainer my priority is to ensure its as easy as possible to see where I need to add new module dependencies (and have that be a single place). If that comes at the cost of bespoke builds requiring multiple configure runs, its a price I'm prepared to pay :)